src/server/cluster_family_test.cc

```diff
@@ -498,6 +498,43 @@ TEST_F(ClusterFamilyTest, ClusterModeSelectNotAllowed) {
   EXPECT_EQ(Run({"select", "0"}), "OK");
 }
 
+TEST_F(ClusterFamilyTest, ClusterFirstConfigCallDropsEntriesNotOwnedByNode) {
+  Run({"debug", "populate", "50000"});
+
+  EXPECT_EQ(Run({"save", "df"}), "OK");
+
+  auto save_info = service_->server_family().GetLastSaveInfo();
+  EXPECT_EQ(Run({"debug", "load", save_info->file_name}), "OK");
+  EXPECT_EQ(CheckedInt({"dbsize"}), 50000);
+
+  EXPECT_EQ(RunAdmin({"dflycluster", "config", R"json(
+      [
+        {
+          "slot_ranges": [
+            {
+              "start": 0,
+              "end": 16383
+            }
+          ],
+          "master": {
+            "id": "abcd1234",
+            "ip": "10.0.0.1",
+            "port": 7000
+          },
+          "replicas": []
+        }
+      ])json"}),
+            "OK");
+
+  time_t start = std::time(NULL);
+  while (CheckedInt({"dbsize"})) {
+    time_t now = std::time(NULL);
+    // Checking deleting the data does not takes more than 5 senconds in the test.
```

Comment:
s/takes/take
s/senconds/seconds
:)

While here, perhaps `absl`-ify this?
I was thinking something along these lines maybe:

```cpp
// Make sure `dbsize` all slots were removed
constexpr absl::Duration kMaxTime = absl::Seconds(5);
absl::Time start = absl::Now();
while (start + kMaxTime < absl::Now()) {
  if (CheckedInt({"dbsize"}) != 0) {
    break;
  }
}
EXPECT_LE(start + kMaxTime, absl::Now());
```